sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -414,14 +455,15 @@ private Response checkError(Response response) throws ChainException {
     return response;
   }
 
-  private URL createEndpoint(String path) throws MalformedURLException {
-    try {
-      URI u = new URI(this.url.toString() + "/" + path);
-      u = u.normalize();
-      return new URL(u.toString());
-    } catch (URISyntaxException e) {
-      throw new MalformedURLException();
+  private void nextURL(int failedIndex) {
+    if (this.urls.size() == 1) {
+      return; // No point contending on the CAS if there's only one URL.
     }
+
+    // A request to the url at failedIndex just failed. Move to the next
+    // URL in the list.
+    int nextIndex = (failedIndex + 1) % this.urls.size();
```

Comment:
Those methods are on the Builder whereas this one is on the Client. Once a Client is constructed, `urls` should be immutable.